### PR TITLE
Cartridge Kubernetes guide on tarantool.io

### PR DESCRIPTION
**Pull request by vanyarock01**

### CODE REVIEW

**Review by veod32 — APPROVED** [review]

**Review by vasiliy-t — APPROVED** [review]

File: doc/cartridge_kubernetes_guide/index.rst [hunk 1]
```diff
@@ -0,0 +1,1016 @@
...
+
+* `tarantool-operator v0.0.8 <https://github.com/tarantool/tarantool-operator/releases/download/tarantool-operator-0.0.8/tarantool-operator-0.0.8.tgz>`_
+* `cartridge v0.0.8 <https://github.com/tarantool/tarantool-operator/releases/download/cartridge-0.0.8/cartridge-0.0.8.tgz>`_.
+
+Next, you need to pack the docker image with ``tarantool-operator``.
+First, let's pull the required version from ``docker.hub``:
+
+.. code-block:: console
+
+   $ docker pull tarantool/tarantool-operator:0.0.6
```
Comment (NickVolynkin) [inline doc/cartridge_kubernetes_guide/index.rst:880]: @vanyarock01 Why verison 0.0.6? Just a paragraph above we mentioned 0.0.8.

File: doc/cartridge_kubernetes_guide/index.rst [hunk 2]
```diff
@@ -0,0 +1,1016 @@
...
+We need to bring the ``tarantool-cartridge`` and ``tarantool-operator``
+charts and the image of your application inside the internal network.
+
+You can download the charts from the following links:
+
+* `tarantool-operator v0.0.8 <https://github.com/tarantool/tarantool-operator/releases/download/tarantool-operator-0.0.8/tarantool-operator-0.0.8.tgz>`_
+* `cartridge v0.0.8 <https://github.com/tarantool/tarantool-operator/releases/download/cartridge-0.0.8/cartridge-0.0.8.tgz>`_.
+
+Next, you need to pack the docker image with ``tarantool-operator``.
+First, let's pull the required version from ``docker.hub``:
```
Comment (NickVolynkin) [inline doc/cartridge_kubernetes_guide/index.rst:876]: Is `docker.hub` some code entity or are we speaking about the [Docker Hub](https://hub.docker.com/)?

File: doc/cartridge_kubernetes_guide/index.rst [hunk 3]
```diff
@@ -0,0 +1,1016 @@
...
+.. NOTE::
+
+   It is important to describe additional containers strictly
+   after the pim-storage container. Otherwise, problems may occur when
+   updating the version of the application.
+
+   By default, Tarantool Kubernetes operator
+   chooses the first one in the list as the application container.
+
+Now, let's start the installation specifying the path to the directory
```
Comment (NickVolynkin) [inline doc/cartridge_kubernetes_guide/index.rst:830]: start the installation **by** specifying the path?

File: doc/cartridge_kubernetes_guide/index.rst [hunk 4]
```diff
@@ -0,0 +1,1016 @@
...
+     repository: "localhost:5000/test-app"
+     tag: "0.1.0-0-g68f6117"
+     pullPolicy: "IfNotPresent"
+   ...
+
+The complete configuration of the ``values.yaml`` can be found in the
+instructions for installing the Tarantool Cartridge application
+described in the guide earlier.
+
+It remains to unpack the Cartridge card:
```
Comment (NickVolynkin) [inline doc/cartridge_kubernetes_guide/index.rst:984]: card → chart?

File: doc/cartridge_kubernetes_guide/index.rst [hunk 5]
```diff
@@ -0,0 +1,1016 @@
...
+If you have a ready-made cluster in the cloud, you can use it. If not, we
+suggest two ways of how to create a local cluster:
+
+* using :ref:`minikube <cartridge_kubernetes_creating_cluster_minikube>`
+* using :ref:`kind <cartridge_kubernetes_creating_cluster_kind>`.
+
+.. _cartridge_kubernetes_creating_cluster_minikube:
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Using *minikube*
```
Comment (NickVolynkin) [inline doc/cartridge_kubernetes_guide/index.rst:178]: Here the name “minicube” is formatted with *italic*, just above it was plain text and on line 52 it was **bold**. This looks inconsistent. Not sure if we already have an established rule of formatting app names, but we probably should have it.

File: doc/cartridge_kubernetes_guide/index.rst [hunk 6]
```diff
@@ -0,0 +1,1016 @@
...
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Building application
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's create a docker image using ``cartridge-cli``:
+
+.. code-block:: console
+
+   $ cartridge pack docker --tag vanyarock01/test-app:0.1.0-0-g68f6117
```
Comment (NickVolynkin) [inline doc/cartridge_kubernetes_guide/index.rst:133]: Where does the `g68f6117` part come from? Is this a shortened hash of the current git commit?

File: doc/cartridge_kubernetes_guide/index.rst [hunk 7]
```diff
@@ -0,0 +1,1016 @@
...
+   `here <https://kubernetes.io/ru/docs/tasks/tools/install-kubectl/>`_.
+
+   .. code-block:: console
+
+      $ kubectl version --client
+      ---
+      Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.0", GitCommit:"2bd9643cee5b3b3a5ecbd3af49d09018f0773c77", GitTreeState:"clean", BuildDate:"2019-09-18T14:36:53Z",     GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
+
+3. **helm** is a package manager for Kubernetes apps. We need the
+   version 3.3.x. Installation instructions can be found
```
Comment (NickVolynkin) [inline doc/cartridge_kubernetes_guide/index.rst:43]: Current version is 3.4.2. Is 3.4.x okay or not?

File: doc/cartridge_kubernetes_guide/index.rst [hunk 8]
```diff
@@ -0,0 +1,1016 @@
...
+   .. code-block:: console
+
+      $ minikube version
+      ---
+      minikube version: v1.12.3
+      commit: 2243b4b97c131e3244c5f014faedca0d846599f5-dirty
+
+5. **kind** (optional) is another tool for creating a local cluster. It
+   can be used instead of the minicube. Installation instructions can be
+   found
```
Comment (NickVolynkin) [inline doc/cartridge_kubernetes_guide/index.rst:65]: Is there a minimal required version of `kind` to use with this guide?

File: doc/cartridge_kubernetes_guide/index.rst [hunk 9]
```diff
@@ -0,0 +1,1016 @@
...
+4. **minikube** is a tool for creating a local Kubernetes cluster. We
+   need the version 1.12 or higher. Installation instructions can be found
+   `here <https://kubernetes.io/ru/docs/tasks/tools/install-minikube/>`_.
+
+   .. code-block:: console
+
+      $ minikube version
+      ---
+      minikube version: v1.12.3
+      commit: 2243b4b97c131e3244c5f014faedca0d846599f5-dirty
```
Comment (NickVolynkin) [inline doc/cartridge_kubernetes_guide/index.rst:61]: Using minikube built from a dirty state is a little weird. Did you build it on your machine?

File: doc/cartridge_kubernetes_guide/index.rst [hunk 10]
```diff
@@ -0,0 +1,1016 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Building application
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's create a docker image using ``cartridge-cli``:
+
+.. code-block:: console
+
+   $ cartridge pack docker --tag vanyarock01/test-app:0.1.0-0-g68f6117
+   ---
```
Comment (NickVolynkin) [inline doc/cartridge_kubernetes_guide/index.rst:134]: The output didn't look like this example for me:
```
cartridge pack docker --tag nickvolynkin/test-app:0.1.0-0-05f2abbc
   • Packing test-app into docker
   • Temporary directory is set to /Users/n.volynkin/.cartridge/tmp/pack-r9vvzsprvt
   • Initialize application dir
   • Build application in /Users/n.volynkin/.cartridge/tmp/pack-r9vvzsprvt/package-files/test-app
   • Building base image test-app-build
   • Build application in test-app-build
   • Remove container...      
   • Application was successfully built
   • Running `cartridge.post-build`
   • Generate VERSION file    
   • Build result image nickvolynkin/test-app:0.1.0-0-05f2abbc
   • Created result image nickvolynkin/test-app:0.1.0-0-05f2abbc
   • Application was successfully packed
```

File: doc/cartridge_kubernetes_guide/index.rst [hunk 11]
```diff
@@ -0,0 +1,1016 @@
...
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Using *minikube*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Create a Kubernetes cluster of version 1.16.4 with 4GB of RAM (recommended):
+
+.. code-block:: console
+
+   $ minikube start --kubernetes-version v1.16.4 --memory 4096
```
Comment (NickVolynkin) [inline doc/cartridge_kubernetes_guide/index.rst:185]: I explicitly started minikube with k8s version 1.16.4. Turns out, that my kubectl is v. 1.19.3, which can be incompatible with the specified k8s version.

```
❗  /usr/local/bin/kubectl is version 1.19.3, which may have incompatibilites with Kubernetes 1.16.4.
    ▪ Want kubectl v1.16.4? Try 'minikube kubectl -- get pods -A'
```
Can we either freeze or unfreeze both k8s and kubectl versions?

File: doc/cartridge_kubernetes_guide/index.rst [hunk 12]
```diff
@@ -0,0 +1,1016 @@
...
+.. NOTE::
+
+   Use the same version of the charts. If you set the ``tarantool-operator``
+   chart to version 0.0.8, set the ``cartridge`` chart to the same version 0.0.8.
+
+Install *tarantool-operator* in the *tarantool* namespace:
+
+.. code-block:: console
+
+   $ helm install tarantool-operator tarantool/tarantool-operator --namespace tarantool --create-namespace --version 0.0.8
```
Comment (NickVolynkin) [inline doc/cartridge_kubernetes_guide/index.rst:281]: Gives me a bunch of these errors:
```
W0107 10:23:23.531224   28894 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
```
Is this error on our side?

File: doc/cartridge_kubernetes_guide/index.rst [hunk 13]
```diff
@@ -0,0 +1,1016 @@
...
+We don't need to know how to perform administrative actions such as
+joining a node or creating a replica set. The operator knows how to do
+this better, and if you set the value for its desired system
+configuration, it begins to bring the cluster to the desired state.
+
+The Tarantool Kubernetes operator itself is the implementation of the Kubernetes
+Operator design pattern. It offers the automation of work with user
+resources using controllers that respond to various events and changes.
+
+The following links can help you understand this pattern:
```
Comment (NickVolynkin) [inline doc/cartridge_kubernetes_guide/index.rst:320]: It's just awesome that you've provided these links!

File: doc/cartridge_kubernetes_guide/index.rst [hunk 14]
```diff
@@ -0,0 +1,1016 @@
...
+.. _cartridge_kubernetes_troubleshooting:
+
+--------------------------------------------------------------------------------
+Troubleshooting
+--------------------------------------------------------------------------------
+
+When creating, updating, or scaling a cluster, errors may occur due to
+lack of physical resources.
+
+Let's examine possible error indications, root causes and solutions.
```
Comment (NickVolynkin) [inline doc/cartridge_kubernetes_guide/index.rst:709]: I made an error in my Lua code and got this:
```
NAME                                  READY   STATUS             RESTARTS   AGE
routers-0-0                           0/1     CrashLoopBackOff   2          3m47s
storages-0-0                          0/1     CrashLoopBackOff   2          3m43s
storages-1-0                          0/1     CrashLoopBackOff   2          3m42s
tarantool-operator-7879d99ccb-gwkb6   1/1     Running            0          36m
```
Is it a common sutiation, should we also mention it?

Comment (NickVolynkin) [reply doc/cartridge_kubernetes_guide/index.rst:709]: https://sysdig.com/blog/debug-kubernetes-crashloopbackoff/

File: doc/cartridge_kubernetes_guide/index.rst (hunk 11 above)
Comment (vanyarock01) [reply doc/cartridge_kubernetes_guide/index.rst:185]: Probably should freeze `kubernetes` and `kubectl` versions at 1.16.X

File: doc/cartridge_kubernetes_guide/index.rst (hunk 12 above)
Comment (vanyarock01) [reply doc/cartridge_kubernetes_guide/index.rst:281]: No, this is not our fault, but it is worth paying attention.

File: doc/cartridge_kubernetes_guide/index.rst (hunk 14 above)
Comment (vanyarock01) [reply doc/cartridge_kubernetes_guide/index.rst:709]: We can add an item to the troubleshooting section. Good observation.

File: doc/cartridge_kubernetes_guide/index.rst (hunk 7 above)
Comment (vanyarock01) [reply doc/cartridge_kubernetes_guide/index.rst:43]: Version 3.4.x is possible, but the version above will most likely not work

File: doc/cartridge_kubernetes_guide/index.rst (hunk 8 above)
Comment (vanyarock01) [reply doc/cartridge_kubernetes_guide/index.rst:65]: I have not tested it, but based on the release notes I think it is v0.6.0

File: doc/cartridge_kubernetes_guide/index.rst (hunk 10 above)
Comment (vanyarock01) [reply doc/cartridge_kubernetes_guide/index.rst:134]: Maybe you have a different version of `cartridge-cli`?

File: doc/cartridge_kubernetes_guide/index.rst (hunk 9 above)
Comment (vanyarock01) [reply doc/cartridge_kubernetes_guide/index.rst:61]: Hmm, yes, that's weird. I'll try to install a cleaner version)

**Pull request merged**